File: Lecture08/OkHttp/app/src/main/java/com/codingblocks/okhttp/MainActivity.java
package com.codingblocks.okhttp;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import java.io.IOException;

import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class MainActivity extends AppCompatActivity {

    TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = findViewById(R.id.tvResult);
        Button btnRequest = findViewById(R.id.btnRequest);

        btnRequest.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                makeNetworkCall();
            }
        });

    }

    private void makeNetworkCall() {

        OkHttpClient client = new OkHttpClient();

        final Request url = new Request.Builder()
                .url("https://api.github.com/search/users?q=harshit")
                .build();

        client.newCall(url).enqueue(new Callback() {
            @Override
            public void onFailure(Call call, IOException e) {
                //If the request failed, retry again
                call.enqueue(this);
                e.printStackTrace();
            }

            @Override
            public void onResponse(Call call, Response response) throws IOException {
                final String result = response.body().string();

                MainActivity.this.runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        textView.setText(result);
                    }
                });
            }
        });
    }
}

File: Lecture09/PokeApi/app/src/main/java/com/codingblocks/pokeapi/MainActivity.kt
package com.codingblocks.pokeapi

import android.content.DialogInterface
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*
import okhttp3.*

import java.io.IOException

class MainActivity : AppCompatActivity() {

    //var is variable, can change
    //val is value, can't change (similar to final)
    val url: String = "https://pokeapi.co/api/v2/pokemon/1"

    var result: Result? = null

    val test = version

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // If saveInstanceState is null, skip this line else get the value
        savedInstanceState?.getString("KEY")

        tvResult.setOnClickListener {
            //Code here
        }

        makeNetworkCall(url)
    }

    public fun testFunction(input: Int) {
        Log.e("TAG", "Hello $input, ${input.inc()}")
    }

    //Single line function
    fun getCount() = 5

    //String templates
    public fun testFunctionTwo(input: Int) =
            Log.e("TAG", "Hello $input, ${input.inc()}")


    //Anonymous inner classes
    private fun makeNetworkCall(url: String) {

        globalFunc("Test")

        val client = OkHttpClient()
        val request = Request.Builder().url(url).build()
        client.newCall(request).enqueue(object : Callback {
            override fun onFailure(call: Call, e: IOException) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            }

            override fun onResponse(call: Call, response: Response) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            }
        })
    }
}
